modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostgisPrimaryKeyFinderTestSetup.java

```diff
@@ -39,11 +39,34 @@ protected void createMetadataTable() throws Exception {
                         + "check (pk_policy in ('sequence', 'assigned', 'autoincrement')))");
     }
 
+    @Override
+    protected void createMetadataView(String name) throws Exception {
+
+        run(
+                "CREATE TABLE tbl_gt_pk_metadata ( "
+                        + "table_schema VARCHAR(32), "
+                        + "table_name VARCHAR(32) NOT NULL, "
+                        + "pk_column VARCHAR(32) NOT NULL, "
+                        + "pk_column_idx INTEGER, "
+                        + "pk_policy VARCHAR(32), "
+                        + "pk_sequence VARCHAR(64),"
+                        + "unique (table_schema, table_name, pk_column),"
+                        + "check (pk_policy in ('sequence', 'assigned', 'autoincrement')))");
+
+        run("CREATE VIEW " + name + " AS SELECT * FROM tbl_gt_pk_metadata");
```

Comment:
Is this actually doing something useful? Unless the table name is quoted, the database is treating the table names as case insensitive.
E.g., try running this in psql:

```
gttest=# create table FOOBAR(id serial);
CREATE TABLE
gttest=# create table foobar(id serial);
ERROR:  relation "foobar" already exists
```

The primary key finder code is not really "case insensitive" regardless, if you tried to create it as quoted, it would not find it (the documentation [never uses quotes](https://docs.geoserver.org/latest/en/user/data/database/primarykey.html#metadata-table-description) around table names for the ``gt_pk_metadata`` either). Try something like this for example:

```
create table "GT_pk_Metadata" ( ... );
```

and I believe the pk finder won't work. 
My suggestion would be to remove the testing for "case insensitive" behavior, it's not really testing anything.

Reply:
Done.